### PR TITLE
Add some stream helpers for dealing with maps

### PR DESCRIPTION
**Pull request by cdeszaq**
Some more stream utility methods that I built while playing in this code. I didn't end up using them, but felt it was a shame to chuck them, so I tossed them into the `StreamUtil` anyways.

### CODE REVIEW

File: fili-core/src/main/java/com/yahoo/bard/webservice/util/StreamUtils.java [hunk 1]
```diff
@@ -145,4 +146,137 @@
                 return castedOperand;
         };
     }
+
+    /**
+     * Build a function that can map a Map from one map to another.
+     *
+     * @param keyMapper  Function to map from the input Map's Entries to the result Map's keys
+     * @param valueMapper  Function to map from the input Map's Entries to the result Map's values
```
Comment (michael-mclawhorn) [inline fili-core/src/main/java/com/yahoo/bard/webservice/util/StreamUtils.java]: keyMapper is the issue with avoiding reduction.

File: fili-core/src/main/java/com/yahoo/bard/webservice/util/StreamUtils.java [hunk 2]
```diff
@@ -145,4 +146,137 @@
...
+    public static <A, B, K, V, R extends Map<A, B>, T extends Map<K, V>> Function<R, T> mapMap(
+            Function<? super Map.Entry<A, B>, ? extends K> keyMapper,
+            Function<? super Map.Entry<A, B>, ? extends V> valueMapper,
+            Supplier<T> resultSupplier
+    ) {
+        return (R input) -> input.entrySet().stream()
+                .collect(
+                        resultSupplier,
+                        (result, entry) -> result.put(keyMapper.apply(entry), valueMapper.apply(entry)),
+                        Map::putAll
```
Comment (michael-mclawhorn) [inline fili-core/src/main/java/com/yahoo/bard/webservice/util/StreamUtils.java]: https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#merge-K-V-java.util.function.BiFunction-

Does what I believe you want it to do.  Basically, have a defaulting BiFunction that just picks left or right.  And then create a version that takes the remapping function as an argument.

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,8 @@ Current
 
 ### Added:
 
+- [Stream utils for dealing with `Map`s](https://github.com/yahoo/fili/pull/114)
```
Comment (QubitPi) [inline CHANGELOG.md]: Shall we add test for `StreamUtils`?

**Comment by cdeszaq** [comment]
I'm not going to get to this any time soon. If I resurface it, it will be later, but not now.

**Pull request closed without merging**